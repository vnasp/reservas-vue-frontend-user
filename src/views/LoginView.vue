<template>
  <main class="container pt-5">
    <p class="text-primary text-center">Sistema de Reservas</p>
    <h3 class="text-center mb-5 text-uppercase fw-bolder fs-1">Y olvídate del resto</h3>
    <p class="text-center">Registrándote en Hostal Patagonia podrás reservar una habitación para ti y tus acompañantes, y disfrutar de una estadía inolvidable con servicios excepcionales y actividades emocionantes.</p>
    <div class="row d-flex justify-content-evenly mt-5">
      <div class="col-12 col-lg-5 border rounded-2 p-4 bg-body-tertiary">
        <h1 class="my-5 text-center"><i class="bi bi-box-arrow-in-right me-1"></i>Inicio de Sesión</h1>
      <form @submit.prevent="handleLogin" class="row g-3">
        <div class="col-12">
          <label for="email" class="form-label">Correo Electrónico</label>
          <input type="email" v-model="datosLogin.email" class="form-control" id="email" placeholder="Ingresa tu e-mail" required>
        </div>
        <div class="col-12">
          <label for="password" class="form-label">Contraseña</label>
          <input type="password" v-model="datosLogin.password" class="form-control" id="password" placeholder="Ingresa tu contraseña" required>
        </div>
        <div class="col-12 text-center">
          <button type="submit" class="btn btn-primary">Iniciar Sesión</button>
        </div>
      </form>
      </div>
    <div class="col-12 col-lg-5 border rounded-2 p-4 bg-body-tertiary">
      <RegisterView/>
    </div>
  </div>
  </main>
</template>

<script>
import { mapActions } from 'vuex';
import RegisterView from './RegisterForm.vue';

export default {
  name: 'LoginView',
  components: {
    RegisterView,
  },
  data() {
    return {
      datosLogin: {
        email: '',
        password: ''
      }
    };
  },
  methods: {
    ...mapActions(['login']),
    async handleLogin() {
      try {
        const user = await this.login(this.datosLogin);
        // Comprobamos si es administrador
        if (user.adminStatus) {
          this.$router.push('/admin/usuarios');
        } else {
          this.$router.push('/mis-reservas');
        }
      } catch (error) {
        this.$swal('Error al iniciar sesión. Por favor, intenta de nuevo.');
        console.error('Error al iniciar sesión:', error);
      }
    }
  }
}
</script>
