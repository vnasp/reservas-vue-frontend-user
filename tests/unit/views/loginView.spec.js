import { shallowMount } from '@vue/test-utils';
import LoginView from '@/views/LoginView.vue';

describe('LoginView.vue', () => {
  it('Crear un snapshot de la vista de Login', () => {
    // Arrange: Configuración inicial
    const wrapper = shallowMount(LoginView);

    // Act: No aplica

    // Assert: Comprobación de que el HTML coincide con el snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });
});