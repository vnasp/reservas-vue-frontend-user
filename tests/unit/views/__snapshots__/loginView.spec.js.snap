// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView.vue Crear un snapshot de la vista de Login 1`] = `
<main class="container pt-5 h-100 d-flex justify-content-center">
  <div class="w-50">
    <h1 class="my-5 text-center"><i class="bi bi-box-arrow-in-right me-1"></i>Inicio de Sesión</h1>
    <form class="row g-3">
      <div class="col-12"><label for="email" class="form-label">Correo Electrónico</label><input type="email" class="form-control" id="email" placeholder="Ingresa tu e-mail" required=""></div>
      <div class="col-12"><label for="password" class="form-label">Contraseña</label><input type="password" class="form-control" id="password" placeholder="Ingresa tu contraseña" required=""></div>
      <div class="col-12 text-center"><button type="submit" class="btn btn-primary">Iniciar Sesión</button></div>
    </form>
  </div>
</main>
`;
